### DISCUSSION TITLE
Access Pinokio via API

### DISCUSSION THREAD

**Discussion by drdoof2019** [6Morpheus6/pinokio-wiki#1] [Ideas]
Hello, 
I wonder is it possible to access (generate content) via API?
Example: I am using Forge library to generate images via Pinokio. It would be great if I can send prompt and it returns generated image whenever it is ready.
Example2: Assume I am using a LLM UI via pinokio. It would be again great if I chat with it via API.

### Thank you for Pinokio.

**Comment by 6Morpheus6** [comment]
@drdoof2019 Generally it is possible to access Pinokio and / or any app in the local network or any app from the internet via cloudflare tunnel.
You can even set your desired port (available in local sharing) and enter a password (only available in public sharing).
You can generally activate this feature in the _**Advanced Settings**_ on the Settings page of Pinokio or with the _**Share**_ button of an app if you only want to share this specific app.
Only apps with script version 1.1 or greater support network and public sharing.

You can find additional information if you scroll down here about [network sharing](https://github.com/pinokiocomputer/pinokio/releases/tag/2.1.14), and here about [public sharing](https://github.com/pinokiocomputer/pinokio/releases/tag/2.0.1) 
These and other new features will be added to the wiki in the near future.

If you have additional questions about that, please open a Discussion in [Pinokio factory discussions](https://github.com/pinokiofactory/factory/discussions)
**Cocktailpeanut** is the dev of Pinokio and can certainly tell you more about it.

Some apps like Automatic1111 offer an api natively. The usage is not different in Pinokio than installed manually.
For Automatic1111 you just needed to add an --api flag to the command line args in webui-user.bat
The project repos of these apps should tell you if they support it. If you have questions on how to activate it with pinokio, e.g. in case you need to add a flag to the launch command, please let me know. In most cases you can do that in _**start.js**_ or _**start.json**_ of such an app.
You find these files in `.\pinokio\api\app.git` (app.git stands for the app you want to use. For Automatic1111 the folder would be called automatic1111.git for example.)
